django/db/backends/sqlite3/features.py

```diff
@@ -61,6 +61,8 @@ class DatabaseFeatures(BaseDatabaseFeatures):
     insert_test_table_with_defaults = 'INSERT INTO {} ("null") VALUES (1)'
     supports_default_keyword_in_insert = False
     supports_unlimited_charfield = True
+    json_set_array_append_requires_length_as_index = True
```

Comment:
I think we also need `supports_partial_json_update` to depend on `supports_json_field` on SQLite.